### ISSUE TITLE
The training data may cross boundary between different articles?

### ISSUE THREAD

**Issue by shendiaomo** [karpathy/nanoGPT#244]
In the code of `train.py`
```python
    x = torch.stack([torch.from_numpy((data[i:i+block_size]).astype(np.int64)) for i in ix])
    y = torch.stack([torch.from_numpy((data[i+1:i+1+block_size]).astype(np.int64)) for i in ix])
```

There may be a situation: the first half of `data[i:i+block_size]` is from one article, and the second half of `data[i:i+block_size]` is from the following article. Is this expected or neglectable? It seems harmful to training.

**Comment by karpathy** [comment]
Yes this is the typical training regime. There is a special END OF TEXT token separating them, so the model is expected to learn that this token separates unrelated documents.
